lib/AST/Expr.cpp

```diff
@@ -716,6 +716,22 @@ bool Expr::canAppendPostfixExpression(bool appendingPostfixOperator) const {
   llvm_unreachable("Unhandled ExprKind in switch.");
 }
 
+Expr *Expr::getMemberChainBase() {
```

Comment:
We could also accomplish this totally dynamically by calling e.g. `UDE->getBase()->getMemberChainBase()` on each of the below, at the expense of traversing the whole chain of references for long chains. We'd be able to get away without storing the `ChainBase` on all of these expressions, though.